### ISSUE TITLE
Cannot Install TypeScript 2.0 on Ubuntu 16.04

### ISSUE THREAD

**Issue by ss108** [Microsoft/TypeScript#10716]
**TypeScript Version:**  nightly (2.0.0-dev.201xxxxx)

**Expected behavior:**

Installing TypeScript via `npm install -g typescript@next` installs TypeScript 2.0.

**Actual behavior:** 

NPM claims that TypeScript 2.0 was installed, but running `tsc --version` says the version is 1.8.10. This is corroborated by the fact that I can't use the easy new way of utilizing type declarations.

**Description**

I tried installing 2.0, but when following the example of using the types for lodash, as specified [here](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/), I got the error that the module "lodash" could not be found. I was confused, but then checked my version of typescript using `tsc --version`; indeed, it said 1.8.10, not 2.0. : (

I uninstalled typescript, and tried to reinstall 2.0 via all three variations: `npm install -g typescript@rc`, `npm install -g typescript@beta`, and `npm install -g typescript@next`. Nothing works; my version according to `tsc --version` is always 1.8.10.  My OS is Ubuntu 16.04.

When I install a typing via, say, `npm install --save @types/lodash`, it works fine.

**Code**
For what it's worth, here's the tsconfig of my test project.

``` json
{
    "compilerOptions": {
     "typeRoots": [
            "./node_modules/@types"
        ],
        "module": "commonjs",
        "outDir": "./build",
        "target": "ES6",
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "preserveConstEnums": true,
        // "sourceRoot": "./src",
        "declaration": true,
        "sourceMap": true
    },
    "compileOnSave": true,
    "exclude": [
        "node_modules"
    ]
}

```

**Comment by DanielRosenwasser** [comment]
After your `npm install -g typescript@rc`, can you take note of the path that npm installed to? It should be printed following the install.

Also let us know what the result of entering `which tsc` is. Can you let us know what each one is?

**Comment by ss108** [comment]
Hey Daniel, 

Thanks for the quick reply.

Here is c/p from my console of the installation.

```
[~]$ sudo npm install typescript@rc -g
/usr/bin/tsc -> /usr/lib/node_modules/typescript/bin/tsc
/usr/bin/tsserver -> /usr/lib/node_modules/typescript/bin/tsserver
/usr/lib
└── typescript@2.0.2 
```

And from `which tsc`:

```
[~]$ which tsc
/usr/local/bin/tsc
```

It looks like tsc on the command line isn't pointing to the right place? It's odd that tsc even works when pointing to the old location, since I uninstalled TypeScript prior to installing the RC.

**Comment by ss108** [comment]
UPDATE: managed to get around this problem by installing TypeScript non-globally, just to the project I want to use it for. I call `tsc` directly from `node_modules/.bin`, and I point VSCode to my local TypeScript installation in the project's `node_modules`.

**Comment by guacachips** [comment]
If I can help you I think it's not really a TypeScript issue. I've the same problem with other npm packages and we are discussing about that [here](https://github.com/npm/npm/issues/13956) (npm issue).

**Comment by cawa-93** [comment]
I fixed it to delete the directory and create a link instead
```
rm -rf /usr/local/lib/node_modules/typescript
ln -s /usr/lib/node_modules/typescript /usr/local/lib/node_modules/typescript
```

**Comment by SenHeng** [comment]
I've just had a similar issue occur and this is how I fixed it. 

TL;DR:  I updated typescript to v.2.0.10 but tsc was still referring to an older node_modules folder. Problem solved by deleting the old node_modules/ folder in `usr/local/lib` and tsc and tsserver in `usr/local/bin` and reinstalling typescript again. 

Long version below.

Latest typescript installed.
```
PC-6952:bin b06729$ npm i -g typescript
/Users/b06729/.nvm/versions/node/v6.9.1/bin/tsc -> /Users/b06729/.nvm/versions/node/v6.9.1/lib/node_modules/typescript/bin/tsc
```

However, tsc version was still 1.8.10.
```
PC-6952:bin b06729$ tsc -v
Version 1.8.10
```

Seems like tsc pointed to an old, globally installed typescript module before we switched to nvm.
```
PC-6952:~ b06729$ which tsc
/usr/local/bin/tsc
```

We moved to NVM about a year ago but I guess we didn't properly purge the old node_modules. There were a few other things besides typescript still in here.
```
PC-6952:bin b06729$ ls /usr/local/lib/node_modules/
typescript
```

I deleted (actually renamed) that folder and also deleted tsc and tsserver in `/usr/local/bin`. Restarted, re-installed typescript again and now everything's fixed.

```
PC-6952:bin b06729$ which tsc
/Users/b06729/.nvm/versions/node/v6.9.1/bin/tsc
PC-6952:bin b06729$ tsc -v
Version 2.0.10
```
